packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -2422,14 +2426,18 @@ typedef PopInvokedCallback = void Function(bool didPop);
 ///
 /// See also:
 ///
-///  * [PopScope], which provides similar functionality in a widget.
+///  * [PopScopeWithResult], which provides similar functionality in a widget.
 ///  * [ModalRoute.registerPopEntry], which unregisters instances of this.
 ///  * [ModalRoute.unregisterPopEntry], which unregisters instances of this.
-abstract class PopEntry {
-  /// {@macro flutter.widgets.PopScope.onPopInvoked}
-  PopInvokedCallback? get onPopInvoked;
-
-  /// {@macro flutter.widgets.PopScope.canPop}
+abstract class PopEntry<T> {
+  /// {@macro flutter.widgets.PopScopeWithResult.onPopInvoked}
+  // This can't be a function getter since dart vm doesn't allow upcasting
```

Comment:
```suggestion
// This can't be a function getter since the Dart VM doesn't allow upcasting
// the generic type of a function getter. This prevents users from declaring
// PopScopeWithResult with any generic type that is subtype of ModalRoute._popEntries.
```